x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx

```diff
@@ -152,14 +152,13 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
       }
     >
   ];
+  const [errorExists, setErrorExists] = useState(false);
```

Comment:
I would call this `isQueryBarValid` and (un)invert the logic as needed; `errorExists` is a bit too generic/ambiguous.